### DISCUSSION TITLE
Replacement for XML parser

### DISCUSSION THREAD

**Discussion by KaiVolland** [geostyler/geostyler#1853] [Q&A] [answered]
We have two repositories which are using the `xml2js` parser to translate between xml and `geostyler-style`:
 - geostyler-sld-parser
 - geostyler-qgis-parser

The `xml2js` parser has several problems. And the activity of the maintainer seems to be limited so that open PullRequests are not commented or more. Therefore we are looking to replace the parser.

So let's chat what options we have.

**Comment by dnlkoch** [comment]
The [fxp](https://github.com/NaturalIntelligence/fast-xml-parser) project looks very promising.

**Reply by KaiVolland** [reply]
Let's check this out with the SLDs that we use in the SLD-Parser.

**Comment by jansule** [comment]
Browser [DOM API](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model) would be my tool of choice

**Reply by chrismayer** [reply]
Then the transformation to JS objects in the GeoStyler format would be handcrafted, right? I don't want to start the discussion now, just wanted to be sure that I got you right.

**Reply by KaiVolland** [reply]
Manual handling can be tricky here. And as discussed on zoom there might be performance issues with bigger files. If the test with the fxp parser fails we will may come back to this later.

**Reply by weskamm** [reply]
which library you wanna use on the nodejs side of things?

**Comment by KaiVolland** [comment, marked as answer]
We're going to try out the [fxp](https://github.com/NaturalIntelligence/fast-xml-parser) project.
